site/content/en/docs/tasks/troubleshooting/troubleshooting_delete_clusterqueue.md

```diff
@@ -0,0 +1,122 @@
+
+For this example, let's assume your ClusterQueue is named `my-cq`.
+
+## Understanding `kubectl delete clusterqueue`
+
+When you run:
+
+```sh
+kubectl delete clusterqueue my-cq
+```
+
+you are initiating the deletion of the ClusterQueue object. 
+However, if there are Workloads still utilizing the ClusterQueue, the command may hang or take some time to complete.
+
+
+## Why does this happen?
+
+Kueue creates a [Workload](/docs/concepts/workload/) object for each Job to track its admission status. 
+If a Workload is admitted, it is associated with a specific ClusterQueue. 
+Kueue uses a [finalizer](https://kubernetes.io/docs/concepts/overview/working-with-objects/finalizers/) 
+called kueue.x-k8s.io/resource-in-use to prevent deletion of the ClusterQueue while resources are still in use. 
+Consequently, a ClusterQueue with this finalizer cannot be deleted until all resources are released.
+
+
+## How to resolve this issue?
+
+To delete a ClusterQueue, first stop it by applying a [StopPolicy](/docs/concepts/cluster_queue/#stoppolicy) with `HoldAndDrain`. 
+This will evict all associated Workloads and release the resources, allowing the ClusterQueue to be deleted without issues.
+
+
+## How to identify the Workloads related to the ClusterQueue?
+
+To find Workloads linked to a specific ClusterQueue, you can use the following steps:
+
+### Using `kueuectl`
+
+{{% alert title="Note" color="primary" %}}
+If you don't have `kueuectl` installed, follow the [installation guide](/docs/reference/kubectl-kueue/installation/).
+{{% /alert %}}
+
+```shell
+kueuectl list workload --clusterqueue my-cq -A
+```
+
+Example output:
+
+```shell
+NAMESPACE   NAME                     JOB TYPE   JOB NAME       LOCALQUEUE   CLUSTERQUEUE    STATUS     POSITION IN QUEUE   AGE
+default     my-job-job-4gk8s-7b737   job        my-job-4gk8s   my-lq        my-cq           ADMITTED                       27m
+default     my-job-job-826hp-caa72   job        my-job-826hp   my-lq        my-cq           ADMITTED                       1h
+```
+
+### Using `kubectl` and `grep`
+
+Alternatively, you can use `kubectl` with `grep`:
+
+```shell
+kueuectl get workload -A | grep my-cq
+```
+
+Example output:
+
+```shell
+default     my-job-4gk8s-7b737   my-lq   my-cq   True                  30m
+default     my-job-826hp-caa72   my-lq   my-cq   True                  30m
+```
+
+{{% alert title="Note" color="primary" %}}
+The grep command only filters lines based on the specified pattern. 
+It might return additional Workloads if the Workload name or LocalQueue name includes the same name as the ClusterQueue.
+{{% /alert %}}
+
+## How to stop the ClusterQueue?
+
+{{% alert title="Note" color="primary" %}}
+If no Workloads are found for the specified ClusterQueue, you can skip the stopping step and proceed directly to deleting the ClusterQueue.
+{{% /alert %}}
+
+### Using `kueuectl`
+
+To stop the ClusterQueue, run:
+
+```shell
+kueuectl stop clusterqueue my-cq
```

Comment:
does this Drain by default?

Reply:
Yes, we have `--keep-already-running` flag (look [here](https://kueue.sigs.k8s.io/docs/reference/kubectl-kueue/commands/kueuectl_stop/kueuectl_stop_clusterqueue/#options.))